### PR TITLE
Adjust display of restart warning message and rebuild button

### PR DESCRIPTION
**Pull request by danlamanna**
The following now holds for enabling and disabling plugins on a
non-restartable server:
- The first time a plugin is enabled or disabled an informational
alert is displayed warning the user that plugins may not take effect
until a restart of Girder
- No additional messages are shown
- The rebuild/restart button is completely hidden

Supersedes #2152, cc @jeffbaumes @brianhelba

### CODE REVIEW

**Comment by jeffbaumes** [comment]
From the code it looks like a refresh or new tab would make the notification appear again, which seems reasonable to me. +1 to no rebuild/restart button at all, and having no other messages.

File: clients/web/src/views/body/PluginsView.js [hunk 1]
```diff
@@ -57,6 +57,7 @@ var PluginsView = View.extend({
     initialize: function (settings) {
         cancelRestRequests('fetch');
         if (settings.all && settings.enabled) {
+            this.cherrypyServer = settings.cherrypyServer || true;
```
Comment (zachmullen) [inline clients/web/src/views/body/PluginsView.js]: Won't this construct make it impossible to set `this.cherrypyServer` to false? I usually use a ternary and `_.has` for cases where the default is true.

File: clients/web/src/views/body/PluginsView.js [hunk 2]
```diff
@@ -127,7 +128,22 @@ var PluginsView = View.extend({
                   }
               }
               $('button.g-rebuild-and-restart').addClass('btn-danger');
-              $('.g-plugin-rebuild-restart-text').addClass('show');
+
+              if (view.cherrypyServer) {
+                  $('.g-plugin-rebuild-restart-text').addClass('show');
+              }
+
+              if (!view.cherrypyServer && !_.has(view, 'displayedCherrypyNotification')) {
```
Comment (zachmullen) [inline clients/web/src/views/body/PluginsView.js]: Probably worthwhile to convert this to an arrow function and ditch the `view` variable.

File: clients/web/src/views/body/PluginsView.js (hunk 1 above)
Comment (danlamanna) [inline clients/web/src/views/body/PluginsView.js]: Yikes, yes (side note, this code branch is entirely unused by Girder+plugins).

**Comment by danlamanna** [comment]
@zachmullen The issues mentioned should be addressed. Better to use the commits as diffs since I re-indented a block.

File: clients/web/src/templates/body/plugins.pug [hunk 3]
```diff
@@ -1,14 +1,10 @@
 .g-body-title Plugins
-  .g-rebuild-container
+  .g-rebuild-container(class=!cherrypyServer ? 'hide' : '')
```
Comment (zachmullen) [inline clients/web/src/templates/body/plugins.pug]: Unless this `!` operator is intentionally used to cast to boolean, we should invert this condition for readability.

File: clients/web/src/views/body/PluginsView.js [hunk 4]
```diff
@@ -113,23 +115,37 @@ var PluginsView = View.extend({
...
+                var plugin = $(event.currentTarget).attr('key');
+                if (state === true) {
+                    this.enabled.push(plugin);
+                } else {
+                    var idx;
+                    while ((idx = this.enabled.indexOf(plugin)) >= 0) {
+                        this.enabled.splice(idx, 1);
+                    }
+                }
+                $('button.g-rebuild-and-restart').addClass('btn-danger');
```
Comment (zachmullen) [inline clients/web/src/views/body/PluginsView.js]: `this.$`

**Pull request merged**